lib/savings/save_money_with_bucket/number_pad.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+
+class KeyPad extends StatelessWidget {
+  final TextEditingController amountController;
+  final Function onChange;
+  final Function delete;
+
+  KeyPad({
+    required this.onChange,
+    required this.amountController,
+    required this.delete,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    double height = MediaQuery.of(context).size.height;
+
+    return Container(
+      margin: EdgeInsets.only(left: 30, right: 30),
+      child: Expanded(
+        child: Column(
+          children: [
+            SizedBox(
+              height: height / 50,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                buttonKey('1', controller: amountController),
+                buttonKey('2', controller: amountController),
+                buttonKey('3', controller: amountController),
+              ],
+            ),
+            SizedBox(
+              height: height / 40,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                buttonKey('4', controller: amountController),
+                buttonKey('5', controller: amountController),
+                buttonKey('6', controller: amountController),
+              ],
+            ),
+            SizedBox(
+              height: height / 40,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                buttonKey('7', controller: amountController),
+                buttonKey('8', controller: amountController),
+                buttonKey('9', controller: amountController),
+              ],
+            ),
+            SizedBox(
+              height: height / 40,
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: [
+                buttonKey('.', controller: amountController),
```

Comment:
The dot should be a widget (`CircleAvatar` with a radius as per the Figma design) that you can center if it needs to reflect the Figma Design.

😂 I think Rudy will love me for this but I like when the UI looks exactly like Designs because I love her designs and the app should reflect exactly that 😂.

Reply:
So the dot needs to be a string and part of the number text input a user can type out in the text field.
As a widget, I don't think it would be possible to make it a part of the text input, thus it won't have any function asides from adding value to the look and feel of the UI.